### PR TITLE
codegen: consolidate id escaping

### PR DESCRIPTION
**Pull request by sougou**
@alainjobart @tirsen @janneku @demmer This change addresses #3297 and #3279.
I still need to write a test for the binlog server. But wanted to make sure you're ok with this approach.

There were a few places where ids were inconsistently escaped.
This change consolidates those places and makes use of the
parser-provided functionality.

### CODE REVIEW

File: go/vt/binlog/binlog_streamer.go [hunk 1]
```diff
@@ -791,7 +782,7 @@ func writeIdentifiesAsSQL(sql *bytes.Buffer, tce *tableCacheEntry, rs *mysql.Row
 		if valueIndex > 0 {
 			sql.WriteString(" AND ")
 		}
-		sql.WriteString(tce.ti.Columns[c].Name.String())
+		sql.Myprintf("%v", tce.ti.Columns[c].Name)
```
Comment (alainjobart) [inline go/vt/binlog/binlog_streamer.go]: No need to escape anything for column names, as we do for table names?

Comment (sougou) [reply go/vt/binlog/binlog_streamer.go]: `Myprintf("%v")` escapes column names. The code path will lead to `node.Format`. I hope I understood your question correctly.

Comment (alainjobart) [reply go/vt/binlog/binlog_streamer.go]: Yes, cool.

**Comment by alainjobart** [comment]
LGTM

File: go/vt/binlog/binlog_streamer.go [hunk 2]
```diff
@@ -773,7 +764,7 @@ func writeValuesAsSQL(sql *bytes.Buffer, tce *tableCacheEntry, rs *mysql.Rows, r
 // writeIdentifiesAsSQL is a helper method to print the identifies as SQL in the
 // provided bytes.Buffer. It also returns the value for the keyspaceIDColumn,
 // and the array of values for the PK, if necessary.
-func writeIdentifiesAsSQL(sql *bytes.Buffer, tce *tableCacheEntry, rs *mysql.Rows, rowIndex int, getPK bool) (sqltypes.Value, []sqltypes.Value, error) {
+func writeIdentifiesAsSQL(sql *sqlparser.TrackedBuffer, tce *tableCacheEntry, rs *mysql.Rows, rowIndex int, getPK bool) (sqltypes.Value, []sqltypes.Value, error) {
```
Comment (demmer) [inline go/vt/binlog/binlog_streamer.go]: Nit: shouldn't this be `writeIdentifiersAsSQL`?

**Pull request merged**